src/http/handler/webfinger.ts

```diff
@@ -0,0 +1,87 @@
+import type { Context as HonoContext, Next } from 'hono';
+
+import type { Account } from 'account/account.entity';
+import type { KnexAccountRepository } from 'account/account.repository.knex';
+import type { Site, SiteService } from 'site/site.service';
+
+const ACCOUNT_RESOURCE_PREFIX = 'acct:';
+const HOST_REGEX = /^([a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)*\.[a-zA-Z0-9]+)$/;
+
+export function createWebFingerHandler(
+    accountRepository: KnexAccountRepository,
+    siteService: SiteService,
+) {
+    /**
+     * Custom webfinger implementation to allow resources hosted on the www
+     * version of a host to resolve to the non-www version of the host
+     *
+     * @see https://github.com/fedify-dev/fedify/blob/main/src/webfinger/handler.ts
+     */
+    return async function handleWebFinger(ctx: HonoContext, next: Next) {
+        const resource = ctx.req.query('resource');
+
+        // We only support custom handling of `acct:` resources - If the
+        // resource is not an `acct:` resource, fallback to the default
+        // webfinger implementation
+        if (!resource || !resource.startsWith(ACCOUNT_RESOURCE_PREFIX)) {
+            return next();
+        }
+
+        const [_, resourceHost] = resource
+            .slice(ACCOUNT_RESOURCE_PREFIX.length)
+            .split('@');
+        if (!resourceHost || !HOST_REGEX.test(resourceHost)) {
+            return next();
+        }
+
+        let site: Site | null = null;
+
+        site = await siteService.getSiteByHost(resourceHost);
+        if (site) {
+            // If a site is found for the resource host, we don't need to
+            // do any custom handling, we can just defer to the default
+            // webfinger implementation
+            return next();
```

Comment:
I _think_ we do want to override here, at least sometimes

I think we need/want the `subject` value to always be `username@site.com` without the www. like below:

| resource | site url is www.site.com | site url is site.com |
| -------- | ------------- | -- |
| `acct:username@www.site.com`  | `username@site.com` | `username@site.com` |
| `acct:username@site.com` | `username@site.com` | `username@site.com` |

But I think by forwarding to the default implementation we get

| resource | site url is www.site.com | site url is site.com |
| -------- | ------------- | -- |
| `acct:username@www.site.com`  | `username@www.site.com` | `username@site.com` |
| `acct:username@site.com` | `username@site.com` | `username@site.com` |

Reply:
I think it's gonna be a pain in the arse but we should add some cucumber tests for the functionality, because that way if we do forward to fedify still, we can catch any regressions from upgrades

Reply:
ok cool, so if we always want `subject` to contain no `www` then we shouldn't forward onto the default handler for this case. Will update 👍

Reply:
> I think it's gonna be a pain in the arse 

Your assessment is correct 😅

I'll look into getting some added though, I agree it would be nice to get some coverage at a higher level